contents/shells/bash.md

```diff
@@ -0,0 +1,173 @@
+<frontmatter>
+  title: Introduction to bash shell
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Linux bash shell 
+
+Author: [Wang Junming](https://github.com/junming403)
+
+Reviewers: [Rahul Rajesh](https://github.com/rrtheonlyone), [Ong Shu Peng](https://github.com/ongspxm), [Jeremy Choo](https://www.github.com/ChooJeremy), [Tan Zhen Yong
+](https://github.com/Xenonym)
+
+## What is the shell?
+
+In general, a computer can be divided into 3 abstraction layers, hardware, kernel and applications. Users cannot control the hardware directly; instructions have to be given through kernel as the kernel is the one that controls the hardware.
+
+<img src="https://cdn-images-1.medium.com/max/1200/1*zEv6mAa2wzHnz4a5uWW4gw.png" alt="shell-kernel" style="width:300px;"/>
+
+<sub>source of image: [What (really) happens when you type ls -l in the shell](https://medium.com/meatandmachines/what-really-happens-when-you-type-ls-l-in-the-shell-a8914950fd73)</sub>
+
+The **shell** is the interface through which we pass instructions to the kernel. These instructions will then be executed through the hardware. In this article, we will look into **Linux bash shell**.
+
+<img src="https://www.howtogeek.com/wp-content/uploads/2017/03/ximg_58c0939c2d487.png.pagespeed.gp+jp+jw+pj+ws+js+rj+rp+rw+ri+cp+md.ic.5lDzHl2vQG.png" alt="shell-kernel" style="width:600px;"/>
+
+<sub>source of image: [The Best Keyboard Shortcuts for Bash](https://www.howtogeek.com/howto/ubuntu/keyboard-shortcuts-for-bash-command-shell-for-ubuntu-debian-suse-redhat-linux-etc/)</sub>
+
+[**bash**](https://www.gnu.org/software/bash/manual/html_node/) stands for **Bourne Again SHell**, an enhanced version of the original Unix Shell program. It has become the standard shell of various linux distributions. 
+
+### Bash scripts
+
+Bash supports a powerful language for writing scripts, which we usually refers to as **bash scripts**.
+
+Here's one example of bash scripts, suppose you have to check if some given file exist periodically. The following script will ask for your input and output the results.
+
+```bash
+#!/bin/bash
+ 
+echo -e "Please input a filename, I will check if the file exists.\n"
+
+# waiting for user input.
+read -p "Input a filename : " filename
+
+# check if file exists.
+test -z $filename && echo "You MUST input a filename." && exit 0
+test ! -e $filename && echo "The file '$filename' DO NOT EXIST" && exit 0
+test -f $filename && filetype="regular file"
+test -d $filename && filetype="directory"
+
+# output result.
+echo "The file: $filename is a $filetype"
+```
```

Comment:
The script has a lot of error handling code. You can consider simplifying the script by removing error handling code. The simpler the script is, the more impressed the reader will be. Anyway, I'll leave that up to you.